p2p/host/resource-manager/obs/stats.go

```diff
@@ -1,72 +1,131 @@
+	}
 
-var (
-	directionTag, _ = tag.NewKey("dir")
-	scopeTag, _     = tag.NewKey("scope")
-	serviceTag, _   = tag.NewKey("service")
-	protocolTag, _  = tag.NewKey("protocol")
-	resourceTag, _  = tag.NewKey("resource")
+	// PeerConns
+	peerConns = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: metricNamespace,
+		Name:      "peer_connections",
+		Buckets:   oneTenThenExpDistributionBuckets,
+		Help:      "Number of connections this peer has",
+	}, []string{"dir"})
+	peerConnsInbound  = peerConns.With(prometheus.Labels{"dir": "inbound"})
+	peerConnsOutbound = peerConns.With(prometheus.Labels{"dir": "outbound"})
+
+	// Lets us build a histogram of our current state. See https://github.com/libp2p/go-libp2p-resource-manager/pull/54#discussion_r911244757 for more information.
+	previousPeerConns = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: metricNamespace,
+		Name:      "previous_peer_connections",
+		Buckets:   oneTenThenExpDistributionBuckets,
+		Help:      "Number of connections this peer previously had. This is used to get the current connection number per peer histogram by subtracting this from the peer_connections histogram",
+	}, []string{"dir"})
+	previousPeerConnsInbound  = previousPeerConns.With(prometheus.Labels{"dir": "inbound"})
+	previousPeerConnsOutbound = previousPeerConns.With(prometheus.Labels{"dir": "outbound"})
+
+	// Streams
+	streams = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: metricNamespace,
+		Name:      "streams",
+		Help:      "Number of Streams",
+	}, []string{"dir", "scope", "protocol"})
+
+	peerStreams = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: metricNamespace,
+		Name:      "peer_streams",
+		Buckets:   oneTenThenExpDistributionBuckets,
+		Help:      "Number of streams this peer has",
+	}, []string{"dir"})
+	peerStreamsInbound  = peerStreams.With(prometheus.Labels{"dir": "inbound"})
+	peerStreamsOutbound = peerStreams.With(prometheus.Labels{"dir": "outbound"})
+
+	previousPeerStreams = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: metricNamespace,
+		Name:      "previous_peer_streams",
+		Buckets:   oneTenThenExpDistributionBuckets,
+		Help:      "Number of streams this peer has",
+	}, []string{"dir"})
+	previousPeerStreamsInbound  = previousPeerStreams.With(prometheus.Labels{"dir": "inbound"})
+	previousPeerStreamsOutbound = previousPeerStreams.With(prometheus.Labels{"dir": "outbound"})
+
+	// Memory
+	memory = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: metricNamespace,
+		Name:      "memory",
+		Help:      "Amount of memory reserved as reported to the Resource Manager",
+	}, []string{"scope", "protocol"})
+
+	// PeerMemory
+	peerMemory = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: metricNamespace,
+		Name:      "peer_memory",
+		Buckets:   memDistribution,
+		// Help:      "Amount of memory reserved as reported to the Resource Manager",
```

Comment:
Any reason this is commented?

Reply:
Oh thanks! This actually was meant as a todo for me. Commented out because it's wrong